torchopt/optim/meta/base.py

```diff
@@ -72,21 +72,22 @@ def step(self, loss: torch.Tensor) -> None:  # pylint: disable=too-many-locals
         ):
             flat_params: TupleOfTensors
             flat_params, container_treespec = pytree.tree_flatten_as_tuple(param_container)  # type: ignore[arg-type]
-            if isinstance(state, UninitializedState):
-                state = self.impl.init(flat_params)
             grads = torch.autograd.grad(
                 loss,
                 flat_params,
                 create_graph=True,
                 allow_unused=True,
             )
-            updates, new_state = self.impl.update(
-                grads,
-                state,
-                params=flat_params,
-                inplace=False,
-            )
-            self.state_groups[i] = new_state
+            with torch.no_grad():
+                if isinstance(state, UninitializedState):
+                    state = self.impl.init(flat_params)
+                updates, new_state = self.impl.update(
+                    grads,
+                    state,
+                    params=flat_params,
+                    inplace=False,
+                )
+                self.state_groups[i] = new_state
```

Comment:
`updates` can be detached from the graph while `new_state` should remain in the graph for explicit gradient computation. We need to add a new test for this. cc @JieRen98